server/models/annotationelement.py

```diff
@@ -296,3 +296,14 @@ def updateElements(self, annotation):
         if time.time() - startTime > 10:
             logger.info('inserted %d elements in %4.2fs' % (
                 len(elements), time.time() - startTime))
+
+    def getElementGroupSet(self, annotation):
+        query = {
+            'annotationId': annotation.get('_annotationId', annotation['_id']),
+            '_version': annotation['_version']
+        }
+        groups = sorted(self.collection.distinct('element.group', filter=query))
+        query['element.group'] = None
+        if self.collection.find_one(query):
```

Comment:
We should probably add an index to the annotationelement collection with `annotationId, _version, element.group`.